staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go

```diff
@@ -777,10 +791,40 @@ func (s StaticTransformers) TransformerForResource(resource schema.GroupResource
 	return transformerFromOverrides(s, resource)
 }
 
+var anyGroupAnyResource = schema.GroupResource{
+	Group:    "*",
+	Resource: "*",
+}
+
 func transformerFromOverrides(transformerOverrides map[schema.GroupResource]value.Transformer, resource schema.GroupResource) value.Transformer {
-	transformer := transformerOverrides[resource]
-	if transformer == nil {
-		return identity.NewEncryptCheckTransformer()
+	if transformer := transformerOverrides[resource]; transformer != nil {
+		return transformer
 	}
-	return transformer
+
+	if transformer := transformerOverrides[schema.GroupResource{
+		Group:    resource.Group,
+		Resource: "*",
+	}]; transformer != nil {
+		return transformer
+	}
+
+	if transformer := transformerOverrides[anyGroupAnyResource]; transformer != nil {
+		return transformer
+	}
+
+	return identity.NewEncryptCheckTransformer()
+}
+
+func grYAMLString(gr schema.GroupResource) string {
```

Comment:
producing "yaml" strings is weird... is this only used for error messages? if so, can we skip the single-quote wrapping and just quote the values in the messages using %q?

Reply:
It's primarily for handling `*.` case. String method of _schema.GroupResource_ parses it to `*` instead of `*.`
```go
if gr.Group == "" && gr.Resource == "*" {
   return "'*.'"
}
```

Reply:
ok, a tweak to emit `*.` is fine, but drop the single-quote wrapping

Reply:
Done.